### ISSUE TITLE
Not possible to purchase a premium theme when signing up with a custom domain

### ISSUE THREAD

**Issue by seear** [Automattic/wp-calypso#21037]
**Steps to reproduce**
1. Starting at URL (logged out of WordPress.com): https://wordpress.com/theme/photo-blog (or any other premium theme)
1. Select "Pick this design" and complete the signup flow.
1. Choose a domain ending in .blog
1. Choose the **Personal** plan
1. Complete the signup flow

**What I expected**
For the theme to be in the cart, then activated on the new site once checkout is complete.

**What happened instead**
No theme in cart. Site starts with a default theme.

**Context/source**
Found while investigating #20717

**Comment by seear** [comment]
From a brief investigation, this seems to be an API or back-end store issue. I see a couple of cart API requests. The first one contains both the theme and the plan, and but the return from that does not contain the theme. The second API request uses that result and consequently the cart contains no theme.

**Comment by scruffian** [comment]
This happens because of another issue, detailed here: p58i-4cf-p2.

**Comment by scruffian** [comment]
This happens with any TLD

**Comment by rachelmcr** [comment]
I tested and confirmed this is still an issue when adding a custom domain with any TLD.

The same flow without a custom domain (premium theme & plan with free site address) works as expected.

**Comment by scruffian** [comment]
Seems to be solved
